databind/src/main/java/gov/nist/secauto/metaschema/databind/io/json/MetaschemaJsonWriter.java

```diff
@@ -307,7 +325,16 @@ private <T extends IFeatureComplexItemValueHandler & IBoundInstanceModelGroupedN
       }
 
       // the field will be the JSON key value
-      String key = jsonKey.getJavaTypeAdapter().asString(keyValue);
+      String key;
+      try {
+        key = jsonKey.getJavaTypeAdapter().asString(keyValue);
+      } catch (IllegalArgumentException ex) {
+        throw new IOException(
+            String.format("Invalid value '%s' for json--key for definition '%s'",
```

Comment:
Minor but did you mean json-key?

```suggestion
            String.format("Invalid value '%s' for json-key for definition '%s'",
```